paddle/fluid/pybind/imperative.cc

```diff
@@ -640,55 +640,109 @@ void BindImperative(py::module *m_ptr) {
-             PADDLE_ENFORCE_EQ(tensor.IsInitialized(), true,
-                               platform::errors::InvalidArgument(
-                                   "%s has not been initialized", self.Name()));
-             return self.NewVarBase(tensor.place(), false);
+           [](const imperative::VarBase
+                  &self) -> std::shared_ptr<imperative::VarBase> {
+             PADDLE_ENFORCE_EQ(
+                 self.Var().IsInitialized() &&
+                     (self.Var().IsType<framework::LoDTensor>() ||
+                      self.Var().IsType<framework::SelectedRows>()),
+                 true, platform::errors::InvalidArgument(
+                           "Tensor %s has not been initialized!", self.Name()));
+             auto detach_var = std::make_shared<imperative::VarBase>(
+                 true, "detach_" + self.Name());
+             detach_var->SetPersistable(self.Persistable());
+             detach_var->SetType(self.Type());
+             detach_var->SetDataType(self.DataType());
+             if (self.Var().IsType<framework::LoDTensor>()) {
+               const auto &origin_tensor =
+                   self.Var().Get<framework::LoDTensor>();
+               PADDLE_ENFORCE_EQ(
+                   origin_tensor.IsInitialized(), true,
+                   platform::errors::InvalidArgument(
+                       "Tensor %s has not been initialized!", self.Name()));
+
+               auto *detach_tensor =
+                   detach_var->MutableVar()->GetMutable<framework::LoDTensor>();
+               detach_tensor->ShareDataWith(origin_tensor);
+             } else {
+               const auto &origin_selected_rows =
+                   self.Var().Get<framework::SelectedRows>();
+               PADDLE_ENFORCE_EQ(
+                   origin_selected_rows.value().IsInitialized(), true,
+                   platform::errors::InvalidArgument(
+                       "Tensor %s has not been initialized!", self.Name()));
+
+               auto *detach_selected_rows =
+                   detach_var->MutableVar()
+                       ->GetMutable<framework::SelectedRows>();
+               detach_selected_rows->set_height(origin_selected_rows.height());
+               detach_selected_rows->set_rows(origin_selected_rows.rows());
+               detach_selected_rows->mutable_value()->ShareDataWith(
+                   origin_selected_rows.value());
+             }
+             VLOG(3) << "The detached Tensor(" << detach_var->Name()
+                     << ") share data with " << self.Name();
+             return detach_var;
            },
-           py::return_value_policy::copy, R"DOC(
+           py::return_value_policy::take_ownership, R"DOC(
 
         Returns a new Tensor, detached from the current graph.
+        It will share data with origin Tensor and always not have a Tensor copy and 
+        Tn addition, the detached Tensor does not provide gradient propagation.
```

Comment:
syntax error.

It will share data with origin Tensor and always not have a Tensor copy and Tn addition, the detached Tensor does not provide gradient propagation.

->

It will share data with origin Tensor and always **does** not have a Tensor copy. **In** addition, the detached Tensor does not provide gradient propagation.

Reply:
Done